File: advanced_javascript/inheritance,super/constructor.js
class Lion {
  constructor(name) {
    this.name = name;
    //생성자 함수 내의 this 는
    //new 를 통해 만들어진 새로운 변수가 된다.
  }
  getName() {
    console.log("내 이름은 " + this.name);
  }
}

myLion = new Lion("King"); //객체 생성
myLion.getName();

File: advanced_javascript/inheritance,super/constructor.py
class Lion:
    def __init__(self, name):
        self.name = name 
        # self는 전달된 객체
        # self.name의 name은 객체변수(인스턴스 변수)
    def getName(self):
        print("내 이름은 " + self.name)

myLion = Lion("King")
myLion.getName()